packages/sdk/src/HopBridge.ts

```diff
@@ -1898,9 +1914,9 @@ class HopBridge extends Base {
     }
     recipient = checksumAddress(recipient)
 
-    let ammWrapper = await this.getAmmWrapper(sourceChain, sourceChain.provider)
-    let l2Bridge = await this.getL2Bridge(sourceChain, sourceChain.provider)
-    const attemptSwapAtSource = this.shouldAttemptSwap(amountOutMin, deadline)
+    const ammWrapper = await this.getAmmWrapper(sourceChain, sourceChain.provider)
+    const l2Bridge = await this.getL2Bridge(sourceChain, sourceChain.provider)
+    const attemptSwapAtSource = this.doesUseAmm && this.shouldAttemptSwap(amountOutMin, deadline)
```

Comment:
I think the `shouldAttemptSwap` method should return false if it uses amm so changing this back to

```suggestion
    const attemptSwapAtSource = this.shouldAttemptSwap(amountOutMin, deadline)
```

 and adding the check here

```ts
  shouldAttemptSwap (amountOutMin: BigNumber, deadline: BigNumberish): boolean {
    if (!this.doesUseAmm) {
       return false
    }
    deadline = BigNumber.from(deadline?.toString() || 0)
    return amountOutMin?.gt(0) || deadline?.gt(0)
  }
```

Reply:
Resolved with 56d91bae233fc3fdecafbecb07dac35a5db04144